fix(operator): Datadog autoscaling how-to (#6555)
Kong/developer.konghq.com: app/_how-tos/operator/operator-dataplane-autoscale-datadog.md

Reviewer: ```suggestion
1. Create a `GatewayConfiguration` that uses it:
```

Reviewer: ```suggestion
1. Patch the `GatewayClass` to use the config:
```

Reviewer: ```suggestion
You can integrate {{ site.operator_product_name }} with [Datadog Metrics](https://docs.datadoghq.com/metrics/) to use {{ site.base_gateway }} latency metrics to autoscale workloads based on their metrics.
```
Revising so it's active voice

       serviceEndpoints: true
   
   agents:
     containers:
       agent:
         env:
         - name: DD_HOSTNAME
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
   ' > values.yaml
   ```

1. Install [Datadog's helm chart](https://github.com/DataDog/helm-charts/tree/main/charts/datadog):

   ```bash
   helm repo add datadog https://helm.datadoghq.com
   helm repo update
   helm install -n default datadog -f values.yaml --set datadog.apiKey=${DD_API_KEY} --set datadog.appKey=${DD_APP_KEY} --set datadog.site=${DD_SITE} datadog/datadog
   ```

1. Wait for the `DatadogMetric` CRD to be established before continuing:

   ```bash
   kubectl wait --for=condition=Established crd/datadogmetrics.datadoghq.com --timeout=120s
   ```

## Send traffic

To trigger autoscaling, run the following command in a new terminal window. This will cause the underlying deployment to sleep for 100ms on each request and thus increase the average response time to that value.

```bash
while curl -k "http://$(kubectl get gateway kong -o custom-columns='name:.status.addresses[0].value' --no-headers -n kong)/command/shell?cmd=sleep%200.1" ; do sleep 1; done
```

> Keep this running while we move on to next steps.

Reviewer: ```suggestion
Keep this running while you move on to the next steps.
```

apiVersion: datadoghq.com/v1alpha1
kind: DatadogMetric
metadata:
  name: command-kong-upstream-latency-ms-avg
  namespace: kong
spec:
  query: autoscaling.kong_upstream_latency_ms{service:command} ' | kubectl apply -f -
```

{:.info}

> > **Note:** The Datadog Cluster Agent only starts refreshing a `DatadogMetric`'s status once it's referenced by a `HorizontalPodAutoscaler`. Its `ACTIVE`/`VALID`/`VALUE` fields will stay empty until you create the HPA in the next section — that's expected, not an error.

Reviewer: ```suggestion
> **Note:** The Datadog Cluster Agent only starts refreshing a `DatadogMetric`'s status once it's referenced by a `HorizontalPodAutoscaler`. Its `ACTIVE`/`VALID`/`VALUE` fields will stay empty until you create the HPA in the next section.
```
